src/lib/Timeline.js

```diff
@@ -343,6 +342,76 @@ export default class ReactCalendarTimeline extends Component {
     windowResizeDetector.removeListener(this)
   }
 
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const {
+      visibleTimeStart,
+      visibleTimeEnd,
+      items,
+      groups
+    } = nextProps
+
+    // This is a gross hack pushing items and groups in to state only to allow
+    // For the forceUpdate check
+    let derivedState = {items, groups}
```

Comment:
well, I understand why you did this but I am not quite sure this is the way to go. I don't have enough knowledge in `getDerivedStateFromProps`, but I think this lifecycle method is enforced by react not to do hacks like this.

I am not very sure what would be the alternative since I am not very well educated on how to use this life methods more than the couple of posts I read about them. I think we should revert back to `componentWillRecieveProps` and migrate to this later on. 

I am interested in knowing what do you think? and I am interested in @mcMickJuice  opinion as well.

Reply:
sorry approved this by mistake

Reply:
The getDerivedStateFromProps is a frustrating lifecycle method. The way I've done this, is the way they intend it to be used.

From React 16.3, `componentWillReceiveProps` is renamed `UNSAFE_componentWillReceiveProps` and this is apparently the way to go.

[See here](https://github.com/facebook/react/issues/12188#issuecomment-364273107)
[and here](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops)